stock_lot_internal_code/models/stock_lot.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class StockLot(models.Model):
+    _inherit = "stock.lot"
+
+    default_code = fields.Char(related="product_id.default_code")
+
+    def name_get(self):
+        name_list = []
+        for rec in self:
+            default_code = rec.default_code
+            if not default_code:
+                name_list.append((rec.id, rec.name))
+                continue
+            name = default_code + " " + rec.name
+            name_list.append((rec.id, name))
+        return name_list
+
+    @api.model
+    def _name_search(
+        self, name, args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        args = list(args or [])
+        if name:
+            args += [
+                "|",
+                ("name", operator, name),
+                ("default_code", operator, name),
+            ]
+        return self._search(args, limit=limit, access_rights_uid=name_get_uid)
```

Comment:
```suggestion
```
Can you check and confirm that this can be replaced with `_rec_names_search = ["name", "product_id.default_code"]`?